File: 백준/Silver/1149. RGB거리/RGB거리.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());

		int[][] arr = new int[n][3];
		int[][] memo = new int[n][3];

		for (int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());

			arr[i][0] = Integer.parseInt(st.nextToken());
			arr[i][1] = Integer.parseInt(st.nextToken());
			arr[i][2] = Integer.parseInt(st.nextToken());
		}

		memo[0][0] = arr[0][0];
		memo[0][1] = arr[0][1];
		memo[0][2] = arr[0][2];

		for (int i = 1; i < n; i++) {
			memo[i][0] = Math.min(memo[i - 1][1], memo[i - 1][2]) + arr[i][0];
			memo[i][1] = Math.min(memo[i - 1][0], memo[i - 1][2]) + arr[i][1];
			memo[i][2] = Math.min(memo[i - 1][0], memo[i - 1][1]) + arr[i][2];
		}

		System.out.println(Math.min(memo[n - 1][0], Math.min(memo[n - 1][1], memo[n - 1][2])));

	}
}

File: 백준/Silver/1149. RGB거리/RGB거리.py
import sys

n = int(sys.stdin.readline().rstrip())

lst = [[0 for _ in range(n)] for _ in range(n)]
memo = [[0 for _ in range(n)] for _ in range(n)]

for i in range(n):
    lst[i] = list(map(int, sys.stdin.readline().rstrip().split()))

memo[0][0] = lst[0][0]
memo[0][1] = lst[0][1]
memo[0][2] = lst[0][2]

for i in range(1, n):
    memo[i][0] = min(memo[i - 1][1], memo[i - 1][2]) + lst[i][0]
    memo[i][1] = min(memo[i - 1][0], memo[i - 1][2]) + lst[i][1]
    memo[i][2] = min(memo[i - 1][1], memo[i - 1][0]) + lst[i][2]

print(min(memo[n - 1][0], memo[n - 1][1], memo[n - 1][2]))
